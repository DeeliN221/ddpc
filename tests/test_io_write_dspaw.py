"""Tests for the write functions in ddpc.io.read.dspaw_as."""

from pathlib import Path

from ddpc.io.read import dspaw_as as rda
from ddpc.io.write import dspaw_as as wda


def test_write(snapshot):
    """Tests reading various as files."""
    for s in Path(f"{__file__}/../structures").resolve().iterdir():
        if s.name.endswith(".as"):
            original_lines = s.read_text().split("\n")
            atoms = rda.read(s)
            write_lines = wda.write("-", atoms)

            for line, write_line in zip(original_lines, write_lines.split("\n"), strict=True):
                # compare them, ignore white spaces
                ori_list = line.split()
                write_list = write_line.split()
                for o, w in zip(ori_list, write_list, strict=True):
                    if o.strip() != w.strip() and abs(float(o.strip()) - float(w.strip())) > 1e-4:
                        raise ValueError(f"{ori_list} != {write_list}")

            assert f"{s.name}\n{atoms}" == snapshot
