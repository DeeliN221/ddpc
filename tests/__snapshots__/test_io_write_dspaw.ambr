# serializer version: 1
# name: test_write
  '''
  mag.as
  Atoms(symbols='Ni2O2', pbc=True, cell=[[4.1684, 2.0842, 2.0842], [2.0842, 4.1684, 2.0842], [2.0842, 2.0842, 4.1684]], initial_magmoms=...)
  '''
# ---
# name: test_write.1
  '''
  magxyz.as
  Atoms(symbols='Ni2O2', pbc=True, cell=[[4.1684, 2.0842, 2.0842], [2.0842, 4.1684, 2.0842], [2.0842, 2.0842, 4.1684]], initial_magmoms=...)
  '''
# ---
# name: test_write.2
  '''
  fixxyz.as
  Atoms(symbols='S4Mo2', pbc=True, cell=[[3.19031572, 0.0, 0.0], [-1.59515786, 2.76289446, 0.0], [0.0, 0.0, 14.87900448]])
  '''
# ---
# name: test_write.3
  '''
  lat.as
  Atoms(symbols='S4Mo2', pbc=True, cell=[[3.19031572, 0.0, 0.0], [-1.59515786, 2.76289446, 0.0], [0.0, 0.0, 14.87900448]])
  '''
# ---
# name: test_write.4
  '''
  all.as
  Atoms(symbols='S4Mo2', pbc=True, cell=[[3.19031572, 0.0, 0.0], [-1.59515786, 2.76289446, 0.0], [0.0, 0.0, 14.87900448]], initial_magmoms=...)
  '''
# ---
# name: test_write.5
  '''
  fixxyzmagxyz.as
  Atoms(symbols='S4Mo2', pbc=True, cell=[[3.19031572, 0.0, 0.0], [-1.59515786, 2.76289446, 0.0], [0.0, 0.0, 14.87900448]], initial_magmoms=...)
  '''
# ---
